### PR TITLE
Amend by default

### PR DESCRIPTION
**Pull request by skjnldsv**
@christophwurst
@georgehrke

### CODE REVIEW

File: lib/git.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ const commitAndPush = async function(path, branch, gitRoot, logger) {
     logger.debug(`Committing assets ${path}`)
 
     git.add(path)
-    await git.commit('Compiling assets', [], { '--signoff': null }, function(err, result) {
+    await git.commit('Compiling assets', [], { '--signoff': null, '--amend': null }, function(err, result) {
```
Comment (ChristophWurst) [inline lib/git.js]: Unrelated, but bothers me :)

```suggestion
    await git.commit('Compile assets', [], { '--signoff': null, '--amend': null }, function(err, result) {
```

Ref https://chris.beams.io/posts/git-commit/#imperative

**Comment by ChristophWurst** [comment]
By default? Can I prevent it?

**Comment by skjnldsv** [comment]
> By default? Can I prevent it?

Well, with this pr, no.
How shall we do it the! Let's discuss this here instead in talk!

**Comment by ChristophWurst** [comment]
with
`./compile amend /js/dist` <- amends
or
`./compile /js/dist ` <- separate commit
or
`./compile fixup /js/dist` <- creates fixup to previous commit

**Comment by skjnldsv** [comment]
Good to me!

**Comment by skjnldsv** [comment]
@ChristophWurst
@georgehrke :D

File: index.js [hunk 2]
```diff
@@ -49,12 +49,24 @@ module.exports = app => {
 			return
 		}
 
-		// 2. command syntax checks
+		// 2.1 command action checks
+		const action = commands.arguments.startsWith('amend')
+			? 'amend'
+			: commands.arguments.startsWith('fixup')
+				? 'fixup'
+				: false
```
Comment (georgehrke) [inline index.js]: Ternary-ception

Comment (skjnldsv) [reply index.js]: So? :p

Comment (ChristophWurst) [reply index.js]: You should add another to make it a ternary ternary.

**Pull request merged**